web/src/main/java/org/cbioportal/web/ClinicalTrialsController.java

```diff
@@ -0,0 +1,175 @@
+package org.cbioportal.web;
+
+import io.swagger.annotations.Api;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContexts;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.cbioportal.web.config.annotation.PublicApi;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+
+@PublicApi
+@RestController
+@Validated
+@Api(tags = "Clinical Trial Data", description = "integrate MolecularMatch to retrieve clinical trials")
+public class ClinicalTrialsController {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClinicalTrialsController.class);
+
+    private String molecularMatchURL;
+
+    @Value("${mm.url:https://api.molecularmatch.com/v1/search/trials}")
+    public void setMolecularMatchURL(String property) {
+
+        this.molecularMatchURL = property;
+    }
+
+
+    @RequestMapping(value = "/molecularmatch", method = RequestMethod.POST)
+    public String getMolecularMatchClinicalTrials(@RequestBody String filters, HttpMethod method) throws
```

Comment:
This function is too long. If possible, please break it up into sub-functions. More comments will also be useful.